classes/class11/README.md

```diff
@@ -1 +1,187 @@
+Before we get started, I'll take a moment to acknowledge one of the authoritative sources for all matters web security, [OWASP](https://www.owasp.org/index.php/Main_Page).
+They've got in-depth explanations and mitigations for most of the known security vulnerabilities, including a handy [top 10 list](https://www.owasp.org/index.php/OWASP_Top_Ten_Cheat_Sheet).
+
+##Common Vulnerabilities
+We'll cover some of the most common vulnerabilities here, but if you're going to be developing a production application, and especially handling any sensitive user data, you'll want to read through OWASP's website and make sure you're not leaving any glaring holes in your application.
+
+##Script Injection
+One of the most common and dangerous attacks on a website is what is known as a script injection.
+This involves the attacker finding a way to get their code executing within the scope of your application, either on your servers, or being served up to your users on the client.
+Let's look at some of the ways that can happen.
+
+###XSS
+A cross-site scripting (XSS) vulnerability is any vulnerability that allows an attacker some means of getting their code to be served by your application to another user.
+How exactly this is accomplished depends on the application, but most commonly it is the result of submitting html or javascript through a form and having your application then render it in a page for someone else to execute.
+
+###eval( )
+We haven't yet talked about how or when to use the `eval()` statement in this class so let's do that now.
+Don't use it.
+`eval()` takes a string as it's argument and will evaluate it as JavaScript code.
+If you avoid using it, that will help mitigate any possibility of arbitrary injected code from running on your servers.
+
+###SQL/NoSQL Injection
+If you've ever worked with a SQL database before you might be able to relate to this XKCD comic:
+
+![Little Bobby Tables](http://imgs.xkcd.com/comics/exploits_of_a_mom.png)
+
+SQL databases are queried with evaluated strings, which, if we're to learn anything from the previous paragraph, can lead to some less than desirable outcomes.
+All it takes for someone to delete or poison your dataset is someone like Little Bobby Tables' mother with a vendetta for against bad application developers.
+
+Now, you might say to yourself "why, we're not using SQL databases, so we must be immune injection attacks. Ha ha ha. Ha ha. Ha."
+That's right, laugh it up.
+Keep laughing right up until you realize that MongoDB is queried with a structured JavaScript API that lends itself to [it's own kind of injection](http://blog.websecurify.com/2014/08/hacking-nodejs-and-mongodb.html), as well as [arbitrary code execution](https://media.blackhat.com/bh-us-11/Sullivan/BH_US_11_Sullivan_Server_Side_WP.pdf) not just against your database, but potentially in your application as well!
```

Comment:
Should we mention that Mongoose (like most/all ORMs) will escape inputs automatically?